### ISSUE TITLE
Type narrowing issue

### ISSUE THREAD

**Issue by mariusmuja** [microsoft/pyright#2119]
**Describe the bug**
Type narrowing doesn't work as expected in cases such as this:
```python
from typing import Union


def bar (a: str):
    pass

def foo(a: Union[str,None]):
    if a == "1" or a=="2":
        bar(a)
```

Pyright give the error:
```
test.py:9:13 - error: Argument of type "Literal['1', '2'] | None" cannot be assigned to parameter "a" of type "str" in function "bar"
    Type "Literal['1', '2'] | None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
```

**Expected behavior**
Type narrowing should infer type of `a` inside the `if` block as `Literal[1,2]`, not `Literal[1,2] | None`.

**VS Code extension or command-line**
Tested with latest pylance version in VSCode and latest pyright version (1.1.158) on the command line.

**Comment by erictraut** [comment]
Thanks for the feature request. I've extended the type narrowing logic for literal comparisons to filter out comparisons with `None` in positive tests.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
